File: app/src/main/java/com/arejas/dashboardofthings/domain/usecases/NetworkManagementUseCase.java
package com.arejas.dashboardofthings.domain.usecases;

import androidx.lifecycle.LiveData;

import com.arejas.dashboardofthings.domain.entities.database.Network;
import com.arejas.dashboardofthings.domain.entities.extended.NetworkExtended;
import com.arejas.dashboardofthings.domain.entities.result.Resource;

import org.jetbrains.annotations.NotNull;

import java.util.List;

public interface NetworkManagementUseCase extends BaseUseCase {

    public LiveData<Resource<List<NetworkExtended>>> getListOfNetworks();

    public LiveData<Resource<NetworkExtended>> getNetwork(@NotNull Integer networkId);

    public LiveData<Resource> createNetwork(@NotNull Network network);

    public LiveData<Resource> updateNetwork(@NotNull Network network);

    public LiveData<Resource> deleteNetwork(@NotNull Network network);

}

File: app/src/main/java/com/arejas/dashboardofthings/domain/usecases/NetworkManagementUseCase.kt
package com.arejas.dashboardofthings.domain.usecases

import androidx.lifecycle.LiveData

import com.arejas.dashboardofthings.domain.entities.database.Actuator
import com.arejas.dashboardofthings.domain.entities.database.Network
import com.arejas.dashboardofthings.domain.entities.database.Sensor
import com.arejas.dashboardofthings.domain.entities.extended.NetworkExtended
import com.arejas.dashboardofthings.domain.entities.result.Resource

interface NetworkManagementUseCase : BaseUseCase {

    val listOfNetworks: LiveData<Resource<List<NetworkExtended>>>

    fun getNetwork(networkId: Int): LiveData<Resource<NetworkExtended>>

    fun getListOfRelatedSensors(networkId: Int): LiveData<Resource<List<Sensor>>>

    fun getListOfRelatedActuators(networkId: Int): LiveData<Resource<List<Actuator>>>

    fun createNetwork(network: Network): LiveData<Resource<*>>

    fun updateNetwork(network: Network): LiveData<Resource<*>>

    fun deleteNetwork(network: Network): LiveData<Resource<*>>

}
